<launch>

  <include file="$(find teo_base)/machines/$(env ROBOT).machines" />

    <!-- icp node -->
    <node pkg="laser_scan_matcher"
               type="laser_scan_matcher_node"
               name="laser_scan_matcher"
               output="screen" >
      <remap from="scan" to="/teo/front_laser/scan" />
      <!-- input <param name="scan_topic" value="/teo/front_laser/scan"/> THIS DOESNT WORK-->
      <!-- output -->
      <!-- frames -->
      <param name="fixed_frame" value="odom"/>
      <param name="base_frame" value="base_footprint"/>
      <!-- parameters -->
      <param name="use_imu" value="false"/>
      <param name="use_odom" value="false"/>
      <param name="use_alpha_beta" value="true"/>
      <param name="max_iterations" value="10"/>
      <!-- publishing -->
      <param name="publish_tf" value="false"/>
      <param name="publish_pose" value="true"/>
    </node>

    <!-- pose2d_odom -->
    <node pkg="iri_pose2d_odom"
               type="pose2d_odom_node"
               name="pose2d_odom"
               output="screen" >
      <!-- input -->
      <remap from="/pose2d_odom/pose2d" to="/pose2D"/>
      <!-- output -->
      <!-- parameters -->
      <param name="parent_id" value="odom"/>
      <param name="frame_id" value="base_footprint"/>
    </node>

</launch>
